### ISSUE TITLE
Dropdown panel should not not be cut off when the editor is inside in an overflow: hidden host

### ISSUE THREAD

**Issue by oleq** [ckeditor/ckeditor5-ui#515]
ATM the dropdown recognizes the boundaries of the viewport and positions accordingly:

![image](https://user-images.githubusercontent.com/1099479/60428953-4b5c9a00-9bfa-11e9-9b10-825f78bbb5aa.png)

![image](https://user-images.githubusercontent.com/1099479/60428960-51527b00-9bfa-11e9-9716-e0b5637b227e.png)

But it does not recognize the `overflow: hidden` ancestors and gets cut off, which is definitely annoying. It makes it impossible to use dropdown in the toolbar in modal windows if the dropdown is to be somewhere later in the toolbar (closer to the right edge).

![image](https://user-images.githubusercontent.com/1099479/60428843-17817480-9bfa-11e9-84ac-88ee30130e0a.png)

**Comment by Reinmar** [comment]
Can it be easily moved to the document collection?

**Comment by oleq** [comment]
I created a similar PoC one day to solve some other issue (don't remember which one) and the biggest issue was that the panel must follow the toolbar when things are scrolled, etc. and there's was some positioning lag we decided was unacceptable. 

Now I remember, I researched the [horizontally scrollable toolbar for mobiles](https://github.com/ckeditor/ckeditor5/issues/416#issuecomment-430246472), gonna quote myself:

----

[...]

* To fight the above, I tried to rewrite dropdown panels to balloon–like components (with `position: absolute` + our positioning heuristics) and I found multiple issues with that approach:
   * The dropdowns were hit by the "position delay syndrome" ![kapture 2018-10-15 at 13 10 45](https://user-images.githubusercontent.com/1099479/47020352-77fa7f80-d159-11e8-85df-1a50c2385ce7.gif)
   * I found out our API is not ready for any UI component using `BalloonPanelView` or anything that goes into the "body" collection. ATM only plugins can use these things because to work with them, we need to know about the editor (`EditorUI`), which is impossible in plain UI components:
       * Where to insert it in DOM (`editor.ui.view.body`)
       * How to handle the focus (`editor.ui.focusTracker`)
   * Because of the above I had to rewrite `createDrodpown` so it gets both the reference to the body collection and the focus tracker so the dropdown can put the (now floating) panel in the right place in DOM and register it in the FC (the panel no longer belongs to the toolbar which is in the FC by default). That's quite a mess.
   * When the dropdown panel is no longer in the same tree as its button (in the "body" collection), when suddenly the toolbar changes its position (e.g. sticky panel is no longer sticky because of focus loss), we must update the position of the panel too. The dropdown has no idea what happens to the toolbar, it doesn't know anything about that (it's encapsulated as a `View`) so there's no way to tell the dropdown to update the position of its panel except from the outside. That would require a very weird API.
![image](https://user-images.githubusercontent.com/1099479/47021092-0e7b7080-d15b-11e8-8503-61997e71104a.png)
![image](https://user-images.githubusercontent.com/1099479/47021106-1509e800-d15b-11e8-8ff4-3ff792c85f02.png)

**Comment by Reinmar** [comment]
It's so sad. In other words – there's no viable solution at the moment. Not only we'd have to rewrite the dropdowns but also toolbars to use exactly the same positioning mechanism.

Am I right that the only solution, for now, is to not use `overflow:hidden`?

**Comment by oleq** [comment]
Not really. We can improve the panel positioning algorithm. It can detect not only the viewport boundaries but also ancestors with `overflow: hidden`. No changes to the DOM structures or toolbars.
